keylime/tpm/tpm_abstract.py

```diff
@@ -277,6 +277,9 @@ def check_pcrs(
     ) -> Failure:
         failure = Failure(Component.PCR_VALIDATION)
 
+        # The measured_boot code loads verifier configuration automatically and therefore cannot be imported globally.
```

Comment:
I am OK, but this suggest a deeper problem in the architecture: how is possible that we depend in an abstract class on something that is in a config file?

The use of the `config` module should be restricted to the cli entry points, that can pass this object or bits of it when down into the tree. This way we resolve problems like this, when a side effect gets uncontrolled with a single `import`

Reply:
> I am OK, but this suggest a deeper problem in the architecture: how is possible that we depend in an abstract class on something that is in a config file?

The main issue is that the registrar and verifier share the abstract TPM code. Once the agent is removed we can refactor it quite easily, because the feature sets used by the registrar and the verifier are disjunct.

Reply:
> how is possible that we depend in an abstract class on something that is in a config file?

The abstract class is having some methods implemented. They could/should probably be moved into tpm_main.py...

Reply:
I think the idea behind tpm_abstract.py was to share code between the TPM 1.2 and TPM 2.0 implementation. 
Now that we only support TPM 2.0 this split does not make much sense anymore. 

Ideally we split the TPM verification part in to its own module similar to the IMA code and build the main event loop so, that we can support different attestation mechanisms.

My idea is to do that before implementing the push model or removing the atomic quote requirement, but after the Python agent is gone, because that makes it easier to implement.